packages/js/product-editor/src/hooks/test/use-error-handler.test.ts

```diff
@@ -108,11 +108,70 @@ describe( 'useErrorHandler', () => {
 			errorProps.actions[ 0 ].onClick();
 		}
 
-		jest.runAllTimers(); // Run all pending timers (for setTimeout)
+		expect( mockFocusByValidatorId ).toHaveBeenCalledWith( 'sku' );
+	} );
+	it( 'should call getParentTabIdByBlockName and focusByValidatorId for product_invalid_sku error when errorProps action is triggered', () => {
+		const error = {
+			code: 'product_form_field_error',
+			validatorId: 'test-validator',
+		} as WPError;
+		const visibleTab = 'general';
+
+		jest.useFakeTimers();
+
+		const { result } = renderHook( () => useErrorHandler() );
+		const { getProductErrorMessageAndProps } = result.current;
+
+		const { errorProps: fieldsErrorProps } = getProductErrorMessageAndProps(
+			error,
+			visibleTab
+		);
+
+		expect( fieldsErrorProps ).toBeDefined();
+		expect( fieldsErrorProps.actions ).toBeDefined();
+		expect( fieldsErrorProps.actions?.length ).toBeGreaterThan( 0 );
+
+		// Trigger the action
+		if ( fieldsErrorProps.actions && fieldsErrorProps.actions.length > 0 ) {
+			fieldsErrorProps.actions[ 0 ].onClick();
+		}
 
-		expect( mockNavigateTo ).toHaveBeenCalledWith( { url: '/new-path' } );
 		expect( mockFocusByValidatorId ).toHaveBeenCalledWith(
 			'test-validator'
 		);
 	} );
+	it( 'should call getParentTabIdByBlockName for product_invalid_sku error when errorProps action is triggered and does not find sku block name', () => {
+		const errors = {
+			code: 'product_invalid_sku',
+		} as WPError;
+		const visibleTab = 'general';
+
+		jest.mock( '../use-blocks-helper', () => ( {
+			useBlocksHelper: jest.fn().mockReturnValue( {
+				getParentTabId: jest.fn( () => 'inventory' ),
+				getParentTabIdByBlockName: jest.fn( () => null ), // Mock returns null
+			} ),
+		} ) );
```

Comment:
This approach doesn't work when you have already `mocked` the path above. Its always good to test that it actually returns the mocked value. In this case `getParentTabIdByBlockName` was still returning `inventory`.
The way to actually mock it is to import the `useBlocksHelper` and calling `mockReturnValue` right on the actual function like so:
```
( useBlocksHelper as jest.Mock ).mockReturnValue( {
	getParentTabId: jest.fn( () => 'inventory' ),
	getParentTabIdByBlockName: jest.fn( () => null ), // Mock returns null
} );
```

Reply:
Fixed in the commit [3a6d054](https://github.com/woocommerce/woocommerce/pull/49729/commits/3a6d05416916cf389393230c9e728a60fb800ac3)